models/classes/user/UserSettingsInterface.php

```diff
@@ -24,9 +24,15 @@
 
 interface UserSettingsInterface
 {
+    public const INTERFACE_MODE = 'interface_mode';
```

Comment:
Lets please add all the settings as constant, so we can start using only the `getSetting` method to get them and remove any temptation of developers to extend UserSettingsInterface with more custom methods per parameters, which is not sustainable for this entity and causes break changes.

Lets please `@deprecate` all other methods except for `get/setSetting`

```suggestion
    public const INTERFACE_MODE = 'interface_mode';
    public const UI_LANGUAGE_CODE = 'ui_language_code';
    public const DATA_LANGUAGE_CODE = 'data_language_code';
    public const TIMEZONE = 'timezone';
```

Reply:
Thanks @Nevermind23 for the fix!